editor/effects.js

```diff
@@ -27,17 +28,30 @@ export default {
 			...edits,
 			content: serialize( getBlocks( state ) ),
 		};
+		const transactionId = uniqueId();
 
 		if ( ! isNew ) {
 			toSend.id = postId;
 		}
 
-		dispatch( { type: 'CLEAR_POST_EDITS' } );
+		dispatch( {
+			type: 'CLEAR_POST_EDITS',
+			optimist: { type: BEGIN, id: transactionId },
+		} );
+		dispatch( {
+			type: 'UPDATE_POST',
+			edits: toSend,
+			optimist: { id: transactionId },
+		} );
 		new wp.api.models.Post( toSend ).save().done( ( newPost ) => {
 			dispatch( {
 				type: 'REQUEST_POST_UPDATE_SUCCESS',
-				post: newPost,
 				isNew,
+				optimist: { type: COMMIT, id: transactionId },
+			} );
+			dispatch( {
+				type: 'UPDATE_POST',
+				edits: newPost,
```

Comment:
I think the "block" state is more close to "edits" than it is to "currentPost". Thus, we shouldn't reset its value after a save success, the user could have updated it.